### PR TITLE
correcting merge errors

### PR DESCRIPTION
**Pull request by dsh33782**
This pull will update dev\main to correct version of src\pz390.java and tests\TESTINS5.MLC that was first attempted on 3/7/22 and got caught in conflict.  I am working on another update based on John G review of prior changes that he completed yesterday before I created this new pull after removing all #300MPY refs.

### CODE REVIEW

**Comment by adelosa** [comment]
Hi Don,
So is this pull request addressing #300? If so, I think you can just keep working and committing to your main branch to address any comments provided by @jyganci from the previous pull request rather than creating a new pull request.  

* @jyganci - maybe hold off on the review of this PR until Don has completed updates to address your previous comments. 
* @abekornelis - can you hold off any further merges to main until we get this change merged just to avoid any issues with conflicts for Don.

**Review by abekornelis — COMMENTED** [review]
I'm with Anthony. Please complete the full set of changes. We will defer any other pull requests that might conflict with these changes until this pull is over and done with...

**Comment by dsh33782** [comment]
I've pushed update today after testing fixes and I think these 2 file updates will fix issues raised by John and after review this pull can be completed so I can move on to other pending fixes for v1.8.01.

**Review by abekornelis — APPROVED** [review]
Since this is 'just' a merge operation, I'd say go ahead and merge. I tried - the build still functions okay, including all automated RT scripts. If there are any hidden issues in this version, they will show up while testing the multiply instructions - they should be in the next pull request. Agreed?

File: tests/TESTINS5.MLC
Comment (jyganci) [inline tests/TESTINS5.MLC:483]: Nothing wrong here. Just suggesting that you move this test up as the last of the other "+/-2,+/-3" tests.

File: tests/TESTINS5.MLC
Comment (jyganci) [inline tests/TESTINS5.MLC:514]: Add "0,3", "3,0" and "0,0" tests as done above for MSGRKC.

File: tests/TESTINS5.MLC
Comment (jyganci) [inline tests/TESTINS5.MLC:545]: Add the three "product is zero" tests as done above for MSGRKC.

**Review by dsh33782 — COMMENTED** [review]
There will be one more pull with update to tests\TESTINS5.MLC to add S0C8 testing suggested by John for overflow on multiply.

**Comment by dsh33782** [comment]
All

 

I am working on an old issue reporting that double ampersands are not
correctly handled by z390 compared to HLASM.

 

Attached is a cut down test program which has 3 different ways to use double
ampersands:

1,  DC C'&&' in called macro

2.  &AMPAMP SETC &&';  DC C'&AMPAM' in called macro

3.  DC  C'&&' in open code

 

The test program gets 3 different answers including one error using current
v1.8.00 version of z390.

Can you help explain what the correct answers are and point me to any
references which help explain the answers?

 

 

Don Higgins

***@***.*** ***@***.***> 

www.don-higgins.net

 


    MACRO
    GENCHR  
    DC C'??' MACRO DC AMPAMP
&AMPAMP SETC '&&'
    DC C'&AMPAMP' MACRO DC WITH SETC AMPAMP
    MEND
    GENCHR
    DC  C'&&' ALC AMPAMP
    END

**Comment by dsh33782** [comment]
Anthony

 

Ok, I hope to have update today or tomorrow and will commit them after testing.

 

Don Higgins

 ***@***.***> ***@***.***

www.don-higgins.net

**Pull request merged**
